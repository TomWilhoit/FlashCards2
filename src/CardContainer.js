import React from "react";
import Card from "./Card";
import "./css/CardContainer.css";

const CardContainer = props => {
  return (
    <div className="CardContainer">
      <Card
        questions={props.questions[props.questionIndex]}
        questionIndex={props.questionIndex}
        incrementQuestionIndex={props.incrementQuestionIndex}
        saveToStorage={props.saveToStorage}
        shouldRepeatQuestions={props.shouldRepeatQuestions}
        restartGame={props.restartGame}
      />
    </div>
  );
};

export default CardContainer;
