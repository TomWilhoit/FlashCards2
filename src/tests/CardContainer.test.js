import React from 'react';
import CardContainer from '../Card';
import { shallow } from 'enzyme';

const questions = {
  "id": 1,
  "question": "What is 'context' in Javascript?",
  "correctAnswer": "The value of the this keyword, in reference to the object that owns the currently executing code",
  "falseAnswer1": "Context is the exact same as scope",
  "falseAnswer2": "Function based reference to the function that 'owns' the currently executing code"
}

describe('CardContainer', () => {
  
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <CardContainer questions={questions}/>
      )
    });
    it('should match snapshot when all data is passed correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  })