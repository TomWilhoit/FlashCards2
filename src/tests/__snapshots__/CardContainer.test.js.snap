// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match snapshot when all data is passed correctly 1`] = `
<div
  className="CardContainer"
>
  <Card
    incrementQuestionIndex={[MockFunction]}
    questionIndex={1}
    restartGame={[MockFunction]}
    saveToStorage={[MockFunction]}
    shouldRepeatQuestions={[MockFunction]}
  />
</div>
`;
